arch/arm/core/cortex_m/mpu/arm_core_mpu.c

```diff
@@ -10,11 +10,99 @@
 #include <soc.h>
 
 #include <arch/arm/cortex_m/mpu/arm_core_mpu_dev.h>
+#include <linker/linker-defs.h>
 
 #define LOG_LEVEL CONFIG_MPU_LOG_LEVEL
 #include <logging/log.h>
 LOG_MODULE_REGISTER(mpu);
 
+
+#if defined(CONFIG_APP_SHARED_MEM)
+#define _MPU_DYNAMIC_REGIONS_AREA_START ((u32_t)&_app_smem_start)
+#else
+#define _MPU_DYNAMIC_REGIONS_AREA_START ((u32_t)&__kernel_ram_start)
+#endif /* CONFIG_APP_SHARED_MEM */
+#define _MPU_DYNAMIC_REGIONS_AREA_SIZE ((u32_t)&__kernel_ram_end - \
+		_MPU_DYNAMIC_REGIONS_AREA_START)
+
+/**
+ * @brief Use the HW-specific MPU driver to program
+ *        the static MPU regions.
+ *
+ * Program the static MPU regions using the HW-specific MPU driver. The
+ * function is meant to be invoked only once upon system initialization.
+ *
+ * If the function attempts to configure a number of regions beyond the
+ * MPU HW limitations, the system behavior will be undefined.
+ *
+ * For some MPU architectures, such as the unmodified ARMv8-M MPU,
+ * the function must execute with MPU enabled.
+ */
+void _arch_configure_static_mpu_regions(void)
+{
+	/* Define a constant array of k_mem_partition objects
+	 * to hold the configuration of the respective static
+	 * MPU regions.
+	 */
+	const struct k_mem_partition static_regions[] = {
+#if defined(CONFIG_APPLICATION_MEMORY)
+		{
+		.start = (u32_t)&__app_ram_start,
+		.size = (u32_t)&__app_ram_end - (u32_t)&__app_ram_start,
+		.attr = K_MEM_PARTITION_P_RW_U_RW,
```

Comment:
The `K_MEM_PARTITION*` defines do not include the cacheability attributes, so we are effectively loosing the TEX, C, B and S bits compare to the previous `_get_mpu_ram_region_attr`.

Reply:
@aurel, the K_MEM_PARTITION_ defines _do_ include the cacheability attributes. When they don't default Cache is assumed. And we also have the _NOCACHE for the __no_cache memory regions.

Reply:
I don't see where the cache attributes are added in ` include/arch/arm/cortex_m/mpu/arm_mpu_v7m.h.` I wanted to give it a try to check and check the values at runtime, but it doesn't even compile:

```
In file included from /home/aurel32/git/zephyrproject/zephyr/include/arch/arm/cortex_m/mpu/arm_mpu.h:13:0,
                 from /home/aurel32/git/zephyrproject/zephyr/include/arch/arm/arch.h:257,
                 from /home/aurel32/git/zephyrproject/zephyr/include/arch/cpu.h:17,
                 from /home/aurel32/git/zephyrproject/zephyr/include/kernel_includes.h:34,
                 from /home/aurel32/git/zephyrproject/zephyr/include/kernel.h:17,
                 from /home/aurel32/git/zephyrproject/zephyr/include/device.h:11,
                 from /home/aurel32/git/zephyrproject/zephyr/arch/arm/core/cortex_m/mpu/arm_core_mpu.c:7:
/home/aurel32/git/zephyrproject/zephyr/arch/arm/core/cortex_m/mpu/arm_core_mpu.c: In function '_arch_configure_static_mpu_regions':
/home/aurel32/git/zephyrproject/zephyr/include/arch/arm/cortex_m/mpu/arm_mpu_v7m.h:219:2: error: invalid operands to binary | (have 'k_mem_partition_attr_t {aka struct <anonymous>}' and 'long unsigned int')
  | NORMAL_OUTER_INNER_NON_CACHEABLE_SHAREABLE)})
  ^
/home/aurel32/git/zephyrproject/zephyr/arch/arm/core/cortex_m/mpu/arm_core_mpu.c:86:11: note: in expansion of macro 'K_MEM_PARTITION_P_RW_U_RW_NOCACHE'
   .attr = K_MEM_PARTITION_P_RW_U_RW_NOCACHE,
           ^
```

Reply:
I workarounded this issue and then compared the configuration of the application memory:

On master:
`[00:00:00.000,000] <dbg> mpu._region_init: [4] 0x20400000 0x130b0016`

With PR#12817:
`[00:00:00.000,013] <dbg> mpu._region_init: [3] 0x20400000 0x13000016`

It means TEX, C and B went from 1 to 0.